doc/libpmemset/pmemset_source_set_extras.3.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: manual
+Content-Style: 'text/css'
+title: _MP(PMEMSET_SOURCE_SET_EXTRAS, 3)
+collection: libpmemset
+header: PMDK
+date: pmemset API version 1.0
+...
+
+[comment]: <> (SPDX-License-Identifier: BSD-3-Clause)
+[comment]: <> (Copyright 2021, Intel Corporation)
+
+[comment]: <> (pmemset_source_set_extras.3 -- man page for pmemset_source_set_extras)
+
+[NAME](#name)<br />
+[SYNOPSIS](#synopsis)<br />
+[DESCRIPTION](#description)<br />
+[RETURN VALUE](#return-value)<br />
+[ERRORS](#errors)<br />
+[SEE ALSO](#see-also)<br />
+
+# NAME #
+
+**pmemset_source_set_extras**() - store extra parameters in the source structure
+
+# SYNOPSIS #
+
+```c
+#include <libpmemset.h>
+
+#define PMEMSET_SDS_DEVICE_ID_LEN ((size_t)512ULL)
+
+struct pmemset_sds {
+	char id[PMEMSET_SDS_DEVICE_ID_LEN];
+	uint64_t usc;
+	int refcount;
+};
+
+struct pmemset_extras {
+	struct pmemset_sds *sds;
+	struct pmemset_badblock *bb;
+	enum pmemset_part_state *state;
+};
+
+struct pmemset_source;
+void pmemset_source_set_extras(struct pmemset_source *src,
+		struct pmemset_extras *ext);
+```
+
+# DESCRIPTION #
+
+The **pmemset_source_set_extras**() stores extra parameters *ext* in the source
+*src* structure.
+
+Shutdown data state parameter *sds* can be initialized for the first time by setting
```

Comment:
There should be an 'shutdown state' man page that explains how to use properly all these pieces together.